fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java

```diff
@@ -835,7 +850,22 @@ private Table resolveTable(TableName tableName) {
             Database database = metadataMgr.getDb(catalogName, dbName);
             MetaUtils.checkDbNullAndReport(database, dbName);
 
-            Table table = metadataMgr.getTable(catalogName, dbName, tbName);
+            Table table = null;
+            if (tableRelation.isSyncMVQuery()) {
+                Pair<Table, MaterializedIndex> materializedIndex =
+                        metadataMgr.getMaterializedViewIndex(catalogName, dbName, tbName);
+                if (materializedIndex != null) {
+                    Table mvTable = materializedIndex.first;
+                    Preconditions.checkState(mvTable != null);
+                    Preconditions.checkState(mvTable instanceof OlapTable);
+                    OlapTable mvOlapTable = ((OlapTable) mvTable).copyOnlyForQuery();
```

Comment:
has concurrent issue. copyOnlyForQuery need lock protection.

Reply:
fix it .